components/browser/state/src/main/java/mozilla/components/browser/state/state/ClosedTabSessionState.kt

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.state
+
+/**
+ * Value type that represents the state of a closed tab.
+ *
+ */
+data class ClosedTabSessionState(
+    var id: Long? = null,
+    var title: String,
+    var url: String,
+    var stateFileName: String,
```

Comment:
I think the room-backed storage implementation is still leaking a bit into this state object: A long id, a file name for a state.

If we use a UUID as an ID (either from the tab or generated) then we can:
* Use this as filename in the storage
* Use this as primary key in room (as an example see the storage for crashes: https://github.com/mozilla-mobile/android-components/blob/master/components/lib/crash/src/main/java/mozilla/components/lib/crash/db/CrashEntity.kt)

If we store the `EngineSessionState` (as mentioned separately) then we can restore this tab and do not need to know the file.